CONTRIBUTING.md

```diff
@@ -145,18 +148,112 @@ make check-pypi-packaging
 - Status checks SHOULD all be green.
   - Reviewer(s) have final word and HAVE TO run tests locally if they merge a PR with a red CI.
 
+#### Git workflow
+
+To keep our git history clean and easy to navigate we rebase our branches before
+merging as mentioned above. `pre-commit` checks if you need to rebase.
```

Comment:
Would be nice to provide solution for https://github.com/packit/pre-commit-hooks/issues/2 here or in the output of the hook. Can you please take a look? Thanks!